fvm/evm/stdlib/contract.cdc

```diff
@@ -601,6 +601,26 @@ contract EVM {
         ) as! Result
     }
 
+    /// Calls a contract function with the given data.
+    /// The execution is limited by the given amount of gas.
+    /// The transaction state changes are not persisted.
+    access(all)
+    fun dryCall(
+        from: EVMAddress,
+        to: EVMAddress,
+        data: [UInt8],
+        gasLimit: UInt64,
+        value: Balance,
+    ): Result {
+        return InternalEVM.dryCall(
```

Comment:
Who is the `msg.sender` and `txn.origin` in this instance? I would expect `dryCall` to be exposed on the COA - `coa.call` and `dryCall` - but maybe that's just a matter of verbiage.

Reply:
> I think exposing it _additionally_ on the COA makes sense for convenience

Ah right, to have it additionally there makes sense indeed. As long as it doesn't force users to create a COA in the first place.

Reply:
I see the benefit in exposing on the EVM contract to remove the dependency on a pre-existing COA. Agreed, adding to COA makes sense as well

Reply:
> I see the benefit in exposing on the EVM contract to remove the dependency on a pre-existing COA. Agreed, adding to COA makes sense as well

@sisyphusSmiling I have included the `coa.dryCall` method as well, in https://github.com/onflow/flow-go/pull/6838/commits/fbd8a293d805e0901e48917d4685ad2a8fbec5c9 . However, I have implicitly used the COA's address, as the `from` value. To keep it consistent with the API of the `coa.call` & `coa.deploy` etc.